spec/src/main/asciidoc/configuration.asciidoc

```diff
@@ -352,15 +341,81 @@ See https://docs.oracle.com/javase/8/docs/api/java/net/URL.html[java.net.URL] ja
+
+If the encrypted content is an inner nested JWT then its signature has to be verified by configuring the verification key as described in the <<signed-jwt-tokens, Signed JWT tokens>> section.
+
+See the <<claims-verification, Verification of JWT token claims>> section how to verify the token claims once the token has been decrypted and the signature of its nested token (if present) verified.
+
+### Decryption Configuration Parameters
+
+#### `mp.jwt.decrypt.key.location`
+
+The `mp.jwt.verify.decryptkey.location` config property allows for an external or internal location
+of Private Decryption Key to be specified.  The value may be a relative path or a URL.
+Please see <<verification-publickey-location, mp.jwt.publickey.location>> for all the information about the supported locations and <<encrypted-jwt-tokens, Encrypted JWT claims and nested tokens>> section for the additional recommendations.
+
+[[claims-verification]]
+## Verification of JWT token claims
+
+MP JWT specification currently supports the verification of the token `iss` issuer and `aud` audience claims which is done after the token signature has been verified or the token has been decrypted.
+
+[[verify-issuer]]
+### `mp.jwt.verify.issuer`
 
 The `mp.jwt.verify.issuer` config property allows for the expected value of the `iss`
 claim to be specified.  A MicroProfile JWT implementation must verify the `iss` claim of incoming JWTs is present and matches the configured value of `mp.jwt.verify.issuer`.
 
-#### `mp.jwt.verify.audiences`
+Note that since this property verifies the `iss` claim value, it will be effective irrespectively of how the token claims have been protected (signed or encrypted or signed first and encrypted next).
+
+### `mp.jwt.verify.audiences`
 
 The `mp.jwt.verify.audiences` config property is a comma delimited list of allowable values for the `aud` claim.  If specified, a MicroProfile JWT implementation must verify the `aud` claim of incoming JWTs is present and at least one value in the claim matches one of the configured values of `mp.jwt.verify.audiences`.
-Note that the aud claim should only be injected into a Set of String, not as a comma delimited string.
+
+Note that the `aud` claim should only be injected into a Set of String, not as a comma delimited string.
+
+## Requirements for accepting signed and encrypted tokens
+
+This specification does not support the application endpoints accepting signed or encrypted or signed and encrypted tokens at the same time. Requiring the implementations to determine a token type at runtime (signed token will have 3 parts, encrypted token - 5 parts) will have a small performance cost and it is expected that a given MP JWT application will accept only signed or only encrypted or only signed and encrypted tokens which will be enforced depending on how `mp.jwt.decrypt.key.location` and either `mp.jwt.verify.publickey.location` or `mp.jwt.verify.publickey` properties are set.
```

Comment:
@rdebusscher I agree it is technically easy to implement, but I'd like the spec to start from the assumption that the the endpoint is not accepting all types of tokens at the same time - because the policy is typically determined in advance - what kind of protection is expected around a token. 
If my requirement is for the application to accept signed then encrypted tokens then if we relax as you suggest and someone sends encrypted-only or signed - only then the application will still accept the tokens despite the requirements being more stringent.

I propose that we keep this restriction for now and if there will be some practical demand we ill easily relax it, we can introduce a property which will users to avoid ambiguities in cases like the one described above (`mp.jwt.token.security=all` or something like that).

Hope it sounds reasonable